patches/chromium/feat_corner_smoothing_css_rule_and_blink_painting.patch

```diff
@@ -0,0 +1,692 @@
+--- a/third_party/blink/renderer/core/paint/box_border_painter.cc
++++ b/third_party/blink/renderer/core/paint/box_border_painter.cc
+@@ -245,7 +245,7 @@ FloatRoundedRect CalculateAdjustedInnerBorder(
+       break;
+   }
+ 
+-  return FloatRoundedRect(new_rect, new_radii);
++  return FloatRoundedRect(new_rect, new_radii, inner_border.Smoothness());
+ }
+ 
+ void DrawSolidBorderRect(GraphicsContext& context,
+@@ -281,7 +281,11 @@ void DrawBleedAdjustedDRRect(GraphicsContext& context,
+       // *outside* of inner rrect, all the way to the layer bounds (enclosing
+       // int rect for the clip, in device space).
+       SkPath path;
+-      path.addRRect(SkRRect(inner));
++      if (inner.IsSmooth()) {
++        path.addPath(inner.SmoothPath());
++      } else {
++        path.addRRect(SkRRect(inner));
++      }
+       path.setFillType(SkPathFillType::kInverseWinding);
+ 
+       cc::PaintFlags flags;
+diff --git a/third_party/blink/renderer/core/paint/box_painter_base.cc b/third_party/blink/renderer/core/paint/box_painter_base.cc
+index 7eb05137650f1221a51f6fbc8277b493078bac68..20af18254c2ca301964859c81bccd6d05e8a4487 100644
+--- a/third_party/blink/renderer/core/paint/box_painter_base.cc
++++ b/third_party/blink/renderer/core/paint/box_painter_base.cc
+@@ -319,7 +319,7 @@ void BoxPainterBase::PaintNormalBoxShadow(const PaintInfo& info,
+     context.SetDrawLooper(draw_looper_builder.DetachDrawLooper());
+ 
+     if (has_border_radius) {
+-      FloatRoundedRect rounded_fill_rect(fill_rect, border.GetRadii());
++      FloatRoundedRect rounded_fill_rect(fill_rect, border.GetRadii(), border.Smoothness());
+       ApplySpreadToShadowShape(rounded_fill_rect, shadow.Spread());
+       context.FillRoundedRect(
+           rounded_fill_rect, Color::kBlack,
+@@ -403,7 +403,7 @@ void BoxPainterBase::PaintInsetBoxShadow(const PaintInfo& info,
+ 
+     gfx::RectF inner_rect = bounds.Rect();
+     AdjustRectForSideClipping(inner_rect, shadow, sides_to_include);
+-    FloatRoundedRect inner_rounded_rect(inner_rect, bounds.GetRadii());
++    FloatRoundedRect inner_rounded_rect(inner_rect, bounds.GetRadii(), bounds.Smoothness());
+     ApplySpreadToShadowShape(inner_rounded_rect, -shadow.Spread());
+     if (inner_rounded_rect.IsEmpty()) {
+       // |AutoDarkMode::Disabled()| is used because |shadow_color| has already
+diff --git a/third_party/blink/renderer/core/paint/rounded_border_geometry.cc b/third_party/blink/renderer/core/paint/rounded_border_geometry.cc
+index 1641321846aeaca69df21125141c722f76b27ebc..14e8794a917fe2f1b6a936a841d21eb13b7667c0 100644
+--- a/third_party/blink/renderer/core/paint/rounded_border_geometry.cc
++++ b/third_party/blink/renderer/core/paint/rounded_border_geometry.cc
+@@ -32,6 +32,20 @@ FloatRoundedRect::Radii CalcRadiiFor(const ComputedStyle& style,
+           : gfx::SizeF());
+ }
+ 
++static float SmoothnessFromLength(const Length& length) {
++  if (length.IsNone()) {
++    return 0.0f;
++  }
++  if (length.HasAuto()) {
```

Comment:
Is this equivalent to the value being `system-ui`?

Reply:
Correct, `system-ui` is mapped to "auto" length. I'll add a code comment to clarify 👍